aya/src/programs/uprobe.rs

```diff
@@ -450,3 +432,307 @@ fn resolve_symbol(path: &str, symbol: &str) -> Result<u64, ResolveSymbolError> {
         Ok(sym.address() - section.address() + offset)
     }
 }
+
+/// Error reading from /proc/pid/maps.
+#[derive(Debug, Error)]
+pub enum ProcMapError {
+    /// Unable to read /proc/pid/maps.
+    #[error(transparent)]
+    Read(io::Error),
+
+    /// Error parsing an integer.
+    #[error(transparent)]
+    ParseInt(#[from] ParseIntError),
+
+    /// Error parsing a line of /proc/pid/maps.
+    #[error("proc map entry parse error")]
+    Parse,
+}
+
+/// The memory maps of a process.
+///
+/// This is read from /proc/`pid`/maps.
+///
+/// The information here may be used to resolve addresses to paths.
+pub struct ProcMap {
```

Comment:
Couldn't this be private for now? I understand that USDT might need it public at
some point, but can we flip it public when we merge that?

I don't think we have anything to lose from having it private for now. Instead
if we make it public now and by the time we merge USDT we need to change API for
whatever reason, then we're going to introduce yet another breaking change.

Reply:
It's pub because @addisoncrump asked for it 😉 
I'm happy to flip it to priv and let them convince you why it should be pub

Reply:
Then I'm going to have to review the API 😋

Reply:
It's priv now. Once this is in someone can open a PR and we can all bikeshed on the API 😆 
My priority is to get the USDT work rebased and reviewable.